docs/standard/native-interop/best-practices.md

```diff
@@ -82,25 +82,35 @@ GUIDs are usable directly in signatures. Many Windows APIs take `GUID&` type ali
 
 ## Blittable types
 
-Blittable types are types that have the same bit-level representation in managed and native code. As such they do not need to be converted to another format to be marshaled to and from native code, and as this improves performance they should be preferred.
+Blittable types are types that have the same bit-level representation in managed and native code. As such they do not need to be converted to another format to be marshaled to and from native code, and as this improves performance they should be preferred. Some types are not blittable but are known to contain blittable contents. These types have similar optimizations as blittable types when they are not contained in another type, but are not considered blittable when in fields of structs or for the purposes of [`UnmanagedCallersOnlyAttribute`](xref:System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute).
 
 **Blittable types:**
 
 - `byte`, `sbyte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`, `single`, `double`
+- structs with fixed layout that only have blittable value types for instance fields
+  - fixed layout requires `[StructLayout(LayoutKind.Sequential)]` or `[StructLayout(LayoutKind.Explicit)]`
+  - structs are `LayoutKind.Sequential` by default
+
+**Types with blittable contents:**
+
 - non-nested one-dimensional arrays of blittable types (for example, `int[]`)
-- structs and classes with fixed layout that only have blittable value types for instance fields
+- classes with fixed layout that only have blittable value types for instance fields
   - fixed layout requires `[StructLayout(LayoutKind.Sequential)]` or `[StructLayout(LayoutKind.Explicit)]`
-  - structs are `LayoutKind.Sequential` by default, classes are `LayoutKind.Auto`
+  - classes are `LayoutKind.Auto` by default
 
 **NOT blittable:**
 
 - `bool`
 
 **SOMETIMES blittable:**
 
-- `char`, `string`
+- `char`
+
+**Types with SOMETIMES blittable contents:**
+
+- `string`
 
-When blittable types are passed by reference, they're simply pinned by the marshaller instead of being copied to an intermediate buffer. (Classes are inherently passed by reference, structs are passed by reference when used with `ref` or `out`.)
+When blittable types are passed by reference with `in`, `ref`, or `out`, or when types with blittable contents are passed by value, they're simply pinned by the marshaller instead of being copied to an intermediate buffer.
```

Comment:
This is only true when the desired `CharSet` matches the runtime encoding (i.e. `CharSet.Unicode`). Unless I am missing what is being stated here.

Reply:
I see you mentioned this below. Perhaps clarify here as well.